### DISCUSSION TITLE
How to do Base64 encoding/decoding?

### DISCUSSION THREAD

**Discussion by kitepad** [c-smile/sciter-js-sdk#120] [Q&A]
In my scenario, client app will upload some small images and store as base64 string in server side. So I want to encode image file content as base64 string in client (without file upload) and decode it as image when browse from server.

**Comment by kitepad** [comment]
I got solution by using  sys.fs.$readfile , encode/decode of @sciter with base64.js.

**Comment by c-smile** [comment]
I've added 

* `btoa(buffer: ArrayBuffer) : string`

  Produces Base64 string from binary data. 

* `atob(base64: string) : ArrayBuffer`

  Restores binary data from Base64 encoded string.  

Will be available in the next build.

**Reply by kitepad** [reply]
Thanks. it's best news for me. Base64.js is ok for text file, but have problems for image file.
